python/ray/serve/_private/replica.py

```diff
@@ -727,10 +727,14 @@ async def perform_graceful_shutdown(self):
         # can skip the wait period.
         if self._user_callable_initialized:
             await self._drain_ongoing_requests()
-            await self._user_callable_wrapper.call_destructor()
 
         await self._metrics_manager.shutdown()
 
+        # We call the destructor last because the replica may not have
```

Comment:
is the ordering between this and metrics manager shutdown not important?

Reply:
I'll restore the original ordering once I change the destructor to [log and return](https://github.com/ray-project/ray/pull/45882#discussion_r1643455688). In any case though, the metrics manager pushes autoscaling info and multiplex info to the controller. I don't think the ordering between that and the destructor is very relevant.